### ISSUE TITLE
No Such Method Error when shading the plugin into my own plugin

### ISSUE THREAD

**Issue by arnhav11** [JitseB/NPCLib#14]
```[15:54:35] [Server thread/ERROR]: Error occurred while enabling AtlasMachinationsAddon v1.0-SNAPSHOT (Is it up to date?)
java.lang.NoSuchMethodError: org.bukkit.Server.getOnlinePlayers()[Lorg/bukkit/entity/Player;
	at com.comphenix.tinyprotocol.TinyProtocol.registerPlayers(TinyProtocol.java:228) ~[?:?]
	at com.comphenix.tinyprotocol.TinyProtocol.<init>(TinyProtocol.java:87) ~[?:?]
	at net.jitse.npclib.listeners.PacketListener$2.<init>(PacketListener.java:61) ~[?:?]
	at net.jitse.npclib.listeners.PacketListener.start(PacketListener.java:61) ~[?:?]
	at net.jitse.npclib.NPCLib.registerInternal(NPCLib.java:67) ~[?:?]
	at net.jitse.npclib.NPCLib.<init>(NPCLib.java:57) ~[?:?]
	at com.mc_atlas.atlasmachinationsaddon.AtlasMachinationsAddon.onEnable(AtlasMachinationsAddon.java:44) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:254) ~[patched_1.13.2.jar:git-Paper-436]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:325) ~[patched_1.13.2.jar:git-Paper-436]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) ~[patched_1.13.2.jar:git-Paper-436]
	at org.bukkit.craftbukkit.v1_13_R2.CraftServer.enablePlugin(CraftServer.java:438) ~[patched_1.13.2.jar:git-Paper-436]
	at org.bukkit.craftbukkit.v1_13_R2.CraftServer.enablePlugins(CraftServer.java:352) ~[patched_1.13.2.jar:git-Paper-436]
	at net.minecraft.server.v1_13_R2.MinecraftServer.l(MinecraftServer.java:613) ~[patched_1.13.2.jar:git-Paper-436]
	at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:575) ~[patched_1.13.2.jar:git-Paper-436]
	at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:435) ~[patched_1.13.2.jar:git-Paper-436]
	at net.minecraft.server.v1_13_R2.DedicatedServer.init(DedicatedServer.java:316) ~[patched_1.13.2.jar:git-Paper-436]
	at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:791) ~[patched_1.13.2.jar:git-Paper-436]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]
```

**Comment by JitseB** [comment]
Hi there, thanks for opening this issue!

Could you first provide me with some information on what version of NPCLib you're using? And how you are shading the JAR?

**Comment by arnhav11** [comment]
Latest version of NPCLib (1.4.1 I think)

I've simply added the plugin itself as a dependency. (This issue did not occur with version 1.3)

**Comment by arnhav11** [comment]
Any updates on this?

**Comment by avanight** [comment]
I've ran into this same issue with NPCLib version 1.4.1 using Paper Spigot 582. The stack trace is identical, and does not reproduce when using NPCLib version 1.3.

**Comment by JitseB** [comment]
I might know how this occurred. Are you @arnhav11 (and @avanight) creating the NPCLib instance before or after the `onEnable` method is called?

My apologies for my late reply. I've been incredibly busy with school.

**Comment by arnhav11** [comment]
No worries.

I was instantiating the NPCLib in the onEnable() method.

**Comment by EazyFTW** [comment]
Same issue. Looks to be something to do with TinyProtocol.

**Comment by arnhav11** [comment]
So, I should probably clarify. I was a bit of a noob and said "shade" when I didn't actually shade the API into my own plugin. I added the plugin version of NPCLib as a dependency. I just shaded (actually this time) the API into my own plugin (I use Maven and I use IntelliJ) and now I'm getting another error:
```[11:32:29] [Server thread/ERROR]: [com.mc_atlas.atlasmachinationsaddon.AtlasMachinationsAddon] [NPCLib] Failed to create NPC. Please report the following stacktrace
java.lang.NullPointerException: null
	at com.mc_atlas.atlasmachinationsaddon.npclib.NPCLib.createNPC(NPCLib.java:80) ~[?:?]
	at com.mc_atlas.atlasmachinationsaddon.npclib.NPCLib.createNPC(NPCLib.java:96) ~[?:?]
	at com.mc_atlas.atlasmachinationsaddon.util.NPCSpawner.spawnTownSteward(NPCSpawner.java:16) ~[?:?]
	at com.mc_atlas.atlasmachinationsaddon.util.DatabaseManager.loadCitizens(DatabaseManager.java:97) ~[?:?]
	at com.mc_atlas.atlasmachinationsaddon.AtlasMachinationsAddon.onEnable(AtlasMachinationsAddon.java:108) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:265) ~[paperclip.jar:git-Paper-"c36f2062"]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:334) ~[paperclip.jar:git-Paper-"c36f2062"]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:412) ~[paperclip.jar:git-Paper-"c36f2062"]
	at org.bukkit.craftbukkit.v1_13_R2.CraftServer.enablePlugin(CraftServer.java:443) ~[paperclip.jar:git-Paper-"c36f2062"]
	at org.bukkit.craftbukkit.v1_13_R2.CraftServer.enablePlugins(CraftServer.java:357) ~[paperclip.jar:git-Paper-"c36f2062"]
	at net.minecraft.server.v1_13_R2.MinecraftServer.l(MinecraftServer.java:608) ~[paperclip.jar:git-Paper-"c36f2062"]
	at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:570) ~[paperclip.jar:git-Paper-"c36f2062"]
	at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:431) ~[paperclip.jar:git-Paper-"c36f2062"]
	at net.minecraft.server.v1_13_R2.DedicatedServer.init(DedicatedServer.java:316) ~[paperclip.jar:git-Paper-"c36f2062"]
	at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:787) ~[paperclip.jar:git-Paper-"c36f2062"]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]
```

**Comment by EazyFTW** [comment]
Hm.. I still get the error using gradle, although it works fine using the 3.0 version of NPCLib.

**Comment by JitseB** [comment]
This is actually something that's caused by the support of 1.7 R4. It's actually frustrating me so much that I'll be porting 1.7 R4 support to a legacy branch of NPCLib.

Should soon be sorted!

**Comment by JitseB** [comment]
Will be solved in the same version as the fix for #17.

**Comment by JitseB** [comment]
Fixed in [release 1.4.2](https://github.com/JitseB/NPCLib/releases/tag/v1.4.2)!
